pkg/argocd/argocd.go

```diff
@@ -532,6 +546,47 @@ func getApplicationType(app *v1alpha1.Application) ApplicationType {
 	}
 }
 
+// getApplicationSourceType returns the source type of the application
+func getApplicationSourceType(app *v1alpha1.Application) v1alpha1.ApplicationSourceType {
+
+	if st, set := app.Annotations[common.WriteBackTargetAnnotation]; set &&
+		strings.HasPrefix(st, common.KustomizationPrefix) {
+		return v1alpha1.ApplicationSourceTypeKustomize
+	}
+
+	if app.Spec.HasMultipleSources() {
```

Comment:
This introduces a regression. The old code would observe the app status which included computed fields for source type. The new code looks at the application spec which in our case does not have any of these fields defined.

Reply:
I don't know if its related but we are using an `ApplicationSet`.

Reply:
This seems to fix the issue for me. 

```go
// getApplicationSourceType returns the source type of the application
func getApplicationSourceType(app *v1alpha1.Application) v1alpha1.ApplicationSourceType {

	if st, set := app.Annotations[common.WriteBackTargetAnnotation]; set &&
		strings.HasPrefix(st, common.KustomizationPrefix) {
		return v1alpha1.ApplicationSourceTypeKustomize
	}

	if app.Spec.HasMultipleSources() {
		for _, st := range app.Status.SourceTypes {
			if st == v1alpha1.ApplicationSourceTypeHelm {
				return v1alpha1.ApplicationSourceTypeHelm
			} else if st == v1alpha1.ApplicationSourceTypeKustomize {
				return v1alpha1.ApplicationSourceTypeKustomize
			} else if st == v1alpha1.ApplicationSourceTypePlugin {
				return v1alpha1.ApplicationSourceTypePlugin
			}
		}
		return v1alpha1.ApplicationSourceTypeDirectory
	}

	return app.Status.SourceType
}
```